src/test/java/ch/astorm/jotlmsg/OutlookMessageMIMETest.java

```diff
@@ -193,6 +193,29 @@ public void plainAndHtmlMail_shouldUseMultiPartAlternative() throws Exception {
         assertTrue(firstBodyPart.getDataHandler().getContentType().startsWith("multipart/alternative"));
     }
 
+    @Test
+    public void plainAndHtmlMailWithInlineAttachments_shouldUseMultiPartRelated() throws Exception {
+        OutlookMessage message = new OutlookMessage();
+        message.setSubject("This is a message");
+        message.setFrom("sender@jotlmsg.com");
+        message.setReplyTo(Arrays.asList("reply1@jotlmsg.com", "reply2@jotlmsg.com"));
+        message.setPlainTextBody("Hello,\n\nThis is a simple message.\n\n.Bye.");
+        message.setHtmlBody("<html><body>Simple body</body></html>");
+        message.addRecipient(OutlookMessageRecipient.Type.TO, "cedric@jotlmsg.com", "Cédric");
+        final OutlookMessageAttachment inlineAttachment = new OutlookMessageAttachment("picture.png", "image/png", new ByteArrayInputStream(new byte[0]));
+        inlineAttachment.setContentId("picture.png");
```

Comment:
Could you make a test without the plain text and with an html part actually using linking the png to show how it works ?

Reply:
Sure, good idea.